### PR TITLE
Add workflow to publish release artifacts

### PR DESCRIPTION
**Pull request by NipunaMadhushan**
## Purpose
Currently release artifacts have been publishing to the s3 bucket using a jenkin job. With this workflow, the process has been migrated to github actions.

## Goals
Since jenkin jobs require a physical machine to run, we can reduce external resources which our release process depends on.

## Fixes
https://github.com/ballerina-platform/ballerina-distribution/issues/2994

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .github/workflows/publish-release-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+          date=`date +%Y-%m-%d`
+          winInstSize=`ls -lh ballerina-$LONG_VERSION-windows-x64.msi | cut -d " " -f5 | sed 's/M/mb/g'`
+          linuxInstSize=`ls -lh ballerina-$LONG_VERSION-linux-x64.deb | cut -d " " -f5 | sed 's/M/mb/g'`
+          rpmInstSize=`ls -lh ballerina-$LONG_VERSION-linux-x64.rpm | cut -d " " -f5 | sed 's/M/mb/g'`
+          macInstSize=`ls -lh ballerina-$LONG_VERSION-macos-x64.pkg | cut -d " " -f5 | sed 's/M/mb/g'`
+
+          fileName=metadata.json
+          echo "{" >> $fileName
+          echo "	\"version\":\"$VERSION\"," >> $fileName
+          echo "	\"short-version\":\"$SHORT_VERSION\"," >> $fileName
```
Comment (keizer619) [inline .github/workflows/publish-release-artifacts.yml:99]: @NipunaMadhushan 
As now there us no short version from swan lake side shall we use same value

File: .github/workflows/publish-release-artifacts.yml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    inputs:
+      ballerina_version:
+        description: 'Ballerina Version'
+        default: '2201.1.1'
+        required: true
+      short_version:
+        description: 'Short Version'
+        default: '2201.1.1'
+        required: true
+      display_version:
```
Comment (keizer619) [inline .github/workflows/publish-release-artifacts.yml:14]: @NipunaMadhushan Currently we only show version in dist list. So we should be able to remove this

File: .github/workflows/publish-release-artifacts.yml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        description: 'Display Version'
+        default: '2201.1.1 (Swan Lake Update 1)'
+        required: true
+      code_name:
```
Comment (keizer619) [inline .github/workflows/publish-release-artifacts.yml:18]: This is also should be able to removed if we short version thing

Comment (NipunaMadhushan) [reply .github/workflows/publish-release-artifacts.yml:14]: @keizer619 do you mean to generate display version using version?

**Review by keizer619 — APPROVED** [review]

**Pull request merged**